lighthouse-core/audits/long-tasks.js

```diff
@@ -0,0 +1,96 @@
+
+const UIStrings = {
+  /** Title of a diagnostic LH audit that provides details on the longest running tasks that occur when the page loads. */
+  title: 'Long main-thread tasks',
+  /** Description of a diagnostic LH audit that shows the user the longest running tasks that occur when the page loads. */
+  description: 'Lists the longest tasks on the main thread, ' +
+    'useful for identifying worst contributors to input delay. ' +
+    '[Learn more](https://web.dev/long-tasks-devtools)',
+  /** [ICU Syntax] Label identifying the number of long-running CPU tasks that occurred while loading a web page. */
+  displayValue: `{itemCount, plural,
+  =1 {1 long task found}
+  other {# long tasks found}
+  }`,
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+class LongTasks extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'long-tasks',
+      scoreDisplayMode: Audit.SCORING_MODES.INFORMATIVE,
+      title: str_(UIStrings.title),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['traces', 'devtoolsLogs'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @param {LH.Audit.Context} context
+   * @return {Promise<LH.Audit.Product>}
+   */
+  static async audit(artifacts, context) {
+    const settings = context.settings || {};
+    const trace = artifacts.traces[Audit.DEFAULT_PASS];
+    const tasks = await MainThreadTasks.request(trace, context);
+    const devtoolsLog = artifacts.devtoolsLogs[LongTasks.DEFAULT_PASS];
+    const networkRecords = await NetworkRecords.request(devtoolsLog, context);
+    const multiplier = settings.throttlingMethod === 'simulate' ?
+      settings.throttling.cpuSlowdownMultiplier : 1;
+
+    const jsURLs = BootupTime.getJavaScriptURLs(networkRecords);
+    // Only consider up to 20 long, top-level (no parent) tasks that have an explicit endTime
+    const longtasks = [...tasks]
+      .filter(t => t.duration >= 50 && !t.unbounded && !t.parent)
+      .sort((a, b) => b.duration - a.duration)
+      .slice(0, 20);
+
+    const results = longtasks.map(task => ({
+      url: BootupTime.getAttributableURLForTask(task, jsURLs),
+      group: task.group.label,
+      start: task.startTime,
+      self: task.selfTime,
+      duration: task.duration * multiplier,
+    }));
+
+    /** @type {LH.Audit.Details.Table['headings']} */
+    const headings = [
+      {key: 'url', itemType: 'url', text: str_(i18n.UIStrings.columnURL)},
+      {key: 'start', itemType: 'ms', granularity: 1, text: str_(i18n.UIStrings.columnStartTime)},
+      {key: 'duration', itemType: 'ms', granularity: 1, text: str_(i18n.UIStrings.columnDuration)},
+    ];
+
+    const tableDetails = Audit.makeTableDetails(headings, results);
+
+    let displayValue;
+    if (results.length > 0) {
+      displayValue = str_(UIStrings.displayValue, {itemCount: results.length});
+    }
+
+    return {
+      score: 1,
```

Comment:
for informative audits we typically `return {notApplicable: true}` when there's nothing to show otherwise they'll show up when there's no action to take.